factory/crypto/cryptoComponentsHandler_test.go

```diff
@@ -1,90 +1,127 @@
-	err := managedCryptoComponents.CheckSubcomponents()
-	require.NoError(t, err)
-}
-
-func TestManagedCryptoComponents_Close(t *testing.T) {
-	t.Parallel()
-
-	managedCryptoComponents := getManagedCryptoComponents(t)
-
-	err := managedCryptoComponents.Close()
-	require.NoError(t, err)
-	multiSigner, errGet := managedCryptoComponents.GetMultiSigner(0)
-	require.Nil(t, multiSigner)
-	require.Equal(t, errors.ErrNilCryptoComponentsHolder, errGet)
-}
-
-func getManagedCryptoComponents(t *testing.T) factory.CryptoComponentsHandler {
-	coreComponents := componentsMock.GetCoreComponents()
-	args := componentsMock.GetCryptoArgs(coreComponents)
-	cryptoComponentsFactory, _ := cryptoComp.NewCryptoComponentsFactory(args)
-	require.NotNil(t, cryptoComponentsFactory)
-	managedCryptoComponents, _ := cryptoComp.NewManagedCryptoComponents(cryptoComponentsFactory)
-	require.NotNil(t, managedCryptoComponents)
-	err := managedCryptoComponents.Create()
-	require.NoError(t, err)
-
-	return managedCryptoComponents
+	t.Run("nil factory should error", func(t *testing.T) {
+		t.Parallel()
+
+		managedCryptoComponents, err := cryptoComp.NewManagedCryptoComponents(nil)
+		require.Equal(t, erdErrors.ErrNilCryptoComponentsFactory, err)
+		require.Nil(t, managedCryptoComponents)
+	})
+	t.Run("invalid args should error", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetCryptoArgs(coreComponents)
+		args.Config.Consensus.Type = "invalid"
+		cryptoComponentsFactory, _ := cryptoComp.NewCryptoComponentsFactory(args)
+		managedCryptoComponents, err := cryptoComp.NewManagedCryptoComponents(cryptoComponentsFactory)
+		require.NoError(t, err)
+		err = managedCryptoComponents.Create()
+		require.Error(t, err)
+		require.Nil(t, managedCryptoComponents.BlockSignKeyGen())
+	})
+	t.Run("pub key mismatch", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetCryptoArgs(coreComponents)
+		args.Config.Consensus.Type = "disabled"
+		cryptoComponentsFactory, _ := cryptoComp.NewCryptoComponentsFactory(args)
+		managedCryptoComponents, err := cryptoComp.NewManagedCryptoComponents(cryptoComponentsFactory)
+		require.NoError(t, err)
+		err = managedCryptoComponents.Create()
+		require.True(t, strings.Contains(err.Error(), erdErrors.ErrPublicKeyMismatch.Error()))
+	})
+	t.Run("should work with activateBLSPubKeyMessageVerification", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		args := componentsMock.GetCryptoArgs(coreComponents)
+		args.ActivateBLSPubKeyMessageVerification = true
+		cryptoComponentsFactory, _ := cryptoComp.NewCryptoComponentsFactory(args)
+		managedCryptoComponents, err := cryptoComp.NewManagedCryptoComponents(cryptoComponentsFactory)
+		require.NoError(t, err)
+		err = managedCryptoComponents.Create()
+		require.NoError(t, err)
+	})
+	t.Run("should work with getters", func(t *testing.T) {
```

Comment:
this errors on `managedCryptoComponents.CheckSubcomponents()` 
doesn't quite works :)

Reply:
added specific tests for CheckSubcomponents and SetMultiSignerContainer